testplan/common/utils/match.py

```diff
@@ -132,14 +219,9 @@ def match(
         end_time = start_time + timeout
         read_mode = "rb"
 
-        # As a convenience, we create the compiled regex if a string was
-        # passed.
-        if not hasattr(regex, "match"):
-            regex = re.compile(regex)
-        if isinstance(regex.pattern, str):
-            read_mode = "r"
+        regex = self._prepare_regexp(regex)
 
-        with open(self.log_path, read_mode) as log:
+        with closing(self.log_stream) as log:
```

Comment:
Is it necessary to open-close the log_stream every time we match? I see the __del__ already do close, is that good enough?

Reply:
That was the original behaviour. Having it in the del is just for a last resort, unbfortunately there are no guaranties that the logmatcher is already garbage collected when the drivers/runtime clean off logfiles in some of the cases. I wanted to be on the safe side here.